### PR TITLE
Fix semantics checker

### PR DESCRIPTION
**Pull request by 20joshuaz**
`make all` works now.

**Issues**
Have not implemented any array semantics checking yet.
I want to have the semantics print out the program like the parser.
~~I also want to add a feature that makes sure that functions return something, i.e., something like this shouldn't compile:~~

```
def foo(none -> number):
  if condition_that_could_be_false:
    return 1
```
FIXED

### CODE REVIEW

File: src/sast.ml [hunk 1]
```diff
@@ -4,18 +4,23 @@ type sexpr = dtype * sx
 and sx = 
     SNumberLit of float
   | SBoolLit of bool
-  | SStringLit of string
   | SCharLit of char
+  | SStringLit of string
```
Comment (Akash-Nayar) [inline src/sast.ml:8]: again

File: src/semant.ml [hunk 2]
```diff
@@ -1,112 +1,194 @@
...
-    | StringLit s -> (String, SStringLit s)
-    | CharLit c -> (Char, SCharLit c)
-    | Id id -> (find_id id (fst tbs), SId id)
-    | Binop (e1, op, e2) -> let (type1, sexpr1) = check_expr tbs e1 in let (type2, sexpr2) = check_expr tbs e2 in
-        let final_type = if type1 = type2 then (
-          match op with
-              (Plus | Minus | Times | IntDiv | Div | Mod) when type1 = Number -> Number
-            | (Eq | Neq | Less | Leq | Greater | Geq) when type1 = Number -> Bool
-            | (And | Or) when type1 = Bool -> Bool
+  let default_capacity = 10 in
```
Comment (Akash-Nayar) [inline src/semant.ml:27]: what's this?

**Review by Akash-Nayar — APPROVED** [review]
genius as always

File: src/semant.ml (hunk 2 above)
Comment (20joshuaz) [reply src/semant.ml:27]: just whenever I create a new hashtable, I need to pass in an initial capacity. any number works, I just chose 10.

**Pull request merged**